### PR TITLE
Fix GhostVertexRemover

### PR DESCRIPTION
**Pull request by mad**
Fix for #1749

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by porunov** [comment]
@mad Due to the problem with test containers, I had to revert the latest commit. Please exclude 2 additional commits from your PR

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/GhostVertexRemover.java [hunk 1]
```diff
@@ -110,11 +110,18 @@ public void process(StaticBuffer key, Map<SliceQuery, EntryList> entries, ScanMe
...
         }
+        v.remove();
         //There should be no more system relations to remove
         metrics.incrementCustom(REMOVED_VERTEX_COUNT);
         metrics.incrementCustom(REMOVED_RELATION_COUNT,removedRelations);
     }
 
+    @Override
+    public void workerIterationEnd(final ScanMetrics metrics) {
+        tx.commit();
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/GhostVertexRemover.java]: Wouldn't it be better to commit in the `process` method? That way, the commit is closer to the mutation and I think that it also makes more sense to have those in the same method.
The documentation for the base method also says:

> Can be used to close any resources held by this job.

which means that I wouldn't expect a commit here.

File: janusgraph-test/src/main/java/org/janusgraph/olap/OLAPTest.java [hunk 2]
```diff
@@ -198,6 +198,12 @@ public void removeGhostVertices() throws Exception {
         assertEquals(1,result.getCustom(GhostVertexRemover.REMOVED_VERTEX_COUNT));
         assertEquals(2,result.getCustom(GhostVertexRemover.REMOVED_RELATION_COUNT));
         assertEquals(0,result.getCustom(GhostVertexRemover.SKIPPED_GHOST_LIMIT_COUNT));
+
+        // Second scan should not find any ghost vertices
+        result = executeScanJob(new GhostVertexRemover(graph));
+        assertEquals(0,result.getCustom(GhostVertexRemover.REMOVED_VERTEX_COUNT));
```
Comment (FlorianHockmann) [inline janusgraph-test/src/main/java/org/janusgraph/olap/OLAPTest.java]: (nitpick) Please add a space after the `,`.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/GhostVertexRemover.java (hunk 1 above)
Comment (mad) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/GhostVertexRemover.java]: Process invoked multiple times for same tx so if we close transaction in intermediate state second process failed.

In the same doc:

> This method is semantically aligned with {@link org.apache.tinkerpop.gremlin.process.computer.VertexProgram#workerIterationEnd(Memory)}

And `workIterationEnd` has implementation that commit tx `BulkLoaderVertexProgram`

**Pull request merged**